lib/querystring.js

```diff
@@ -230,7 +229,7 @@ QueryString.parse = QueryString.decode = function(qs, sep, eq, options) {
 
   var keys = [];
   for (var i = 0; i < len; ++i) {
-    const x = qs[i].replace(regexp, '%20');
+    const x = replacePlus(qs[i]);
```

Comment:
Have you tried moving the regexp out of the function so it doesn't keep getting initialized over and over?